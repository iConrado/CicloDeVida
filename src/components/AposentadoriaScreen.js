import React from 'react';
import { 
  Image,
  ScrollView, 
  View, 
  Text, 
  TextInput, } from 'react-native';

import styles from './functions/styles';
import Cabecalho from './functions/Cabecalho';
import Rodape from './functions/Rodape';
import EstiloVoltar from './functions/EstiloVoltar';
import ModalErro from './functions/ModalErro';
import LimiteDeErro from './functions/LimiteDeErro';
import Controle from './functions/Controle';
import Ciclo from './functions/Ciclo';
import monetizar from './functions/monetizar';
import desmonetizar from './functions/desmonetizar';

import SliderTaxa from './aposentadoria/SliderTaxa';
import SliderIdade from './aposentadoria/SliderIdade';
import SliderDisponib from './aposentadoria/SliderDisponib';

const star = require('../imgs/ic_stars_white.png');

const C = new Ciclo();
let objErro = {};
const tmpComprometimento = [];

export default class AposentadoriaScreen extends React.Component {
  static navigationOptions = { //eslint-disable-line
    headerTitle: Cabecalho('Consultoria Ciclo de Vida'),
    headerBackTitle: 'Voltar',
    headerTintColor: EstiloVoltar.hTintColor,
    headerStyle: EstiloVoltar.hStyle,
  };
  
  constructor(props) {
    super(props);
    this.state = {
      modalErro: false, 
      disponib: Math.ceil(this.rendaPercentual() / 50) * 50,
      reservaPrev: 0,
      idadeAposent: 50,
      rentab: 0,
      comprometimento: 0,
    };
    this.fechaErro = this.fechaErro.bind(this);
    this.abreErro = this.abreErro.bind(this);
    this.proxTela = this.proxTela.bind(this);

    this.defRentab = this.defRentab.bind(this);
    this.defIdadeAposent = this.defIdadeAposent.bind(this);
    this.defDisponib = this.defDisponib.bind(this);
  }

  componentWillMount() {
    tmpComprometimento[0] = this.state.disponib;
    this.comprometimentoAtual();
  }

  abreErro(e, tipo) {
    objErro = e;
    this.setState({ modalErro: true });
    switch (tipo) {
      case 0:
        console.log('Retorno 0');
        break;
      default:
        console.log('Retorno default');
    }
  }

  fechaErro() {
    this.setState({ modalErro: false });
    objErro = {};
  }

  faixaEtaria() {
    const faixaEtaria = C.faixaEtaria();

    return faixaEtaria;    
  }

  sugestaoReserva() {
    const sugestaoReserva = C.sugestReserv();

    return sugestaoReserva;
  }

  percentualRenda(valor) {
    if (valor === 0) { return 0; }
    
    const salario = C.getSalLiq();
    const perc = (valor / salario) * 100;

    return perc.toFixed(1).replace('.', ',');
  }

  rendaPercentual() {
    const perc = this.sugestaoReserva();

    if (perc === 0) { return 0; }

    const salario = C.getSalLiq();
    const valor = (perc * salario);

    return parseInt(valor, 10);    
  }

  reservaTotal() {
    const disponib = this.state.disponib;
    const reservaPrev = this.state.reservaPrev;
    const rentab = this.state.rentab;
    const rentabMM = C.taxaMensal(rentab);
    const idadeAtual = C.idadeAtual();
    const idadeAposent = this.state.idadeAposent;
    const prazoAA = idadeAposent - idadeAtual;
    const prazoMM = prazoAA * 12;

    const reservaTotal = C.montante(reservaPrev, disponib, prazoMM, rentabMM);

    return parseInt(reservaTotal, 10);
  }

  defDisponib(valor) {
    this.setState({ disponib: valor });
    tmpComprometimento[0] = valor;
    this.comprometimentoAtual();
  }

  defIdadeAposent(valor) {
    this.setState({ idadeAposent: valor });
  }

  defRentab(valor) {
    this.setState({ rentab: valor });
  }

  comprometimentoAtual() {
    if (tmpComprometimento[0] !== undefined) {
      const valor = tmpComprometimento.reduce((prevVal, elem) => prevVal + elem);
      const compr = C.comprometimentoAtual('Aposentadoria', valor);
      this.setState({ comprometimento: compr });
    }
  }

  proxTela(tela) {
    // Função que valida os campos e submete os dados para registro na classe de negócio.
    // Em caso de algum retorno com erro, executa a abertura da tela de erros.
    const { navigate } = this.props.navigation;

    const disponib = this.state.disponib;
    const reservaPrev = this.state.reservaPrev;
    const idadeAposent = this.state.idadeAposent;
    const rentab = this.state.rentab;

    // Validação das regras de negócio, registro e gravação de log
    if (!Controle(this.abreErro, C, C.setDisponib, disponib)) { return false; }
    if (!Controle(this.abreErro, C, C.setReservaPrev, reservaPrev)) { return false; }
    if (!Controle(this.abreErro, C, C.setIdadeAposent, idadeAposent)) { return false; }
    if (!Controle(this.abreErro, C, C.setRentab, rentab)) { return false; }

    navigate(tela);
  }

  render() {
    return (
      <View style={styles.tela}>
        <ScrollView 
          style={styles.scroll}
          contentContainerStyle={styles.container}
        >
          {/* Camada Modal que intercepta erros e exibe uma mensagem personalizada na tela */}
          <ModalErro 
            visivel={this.state.modalErro}
            fechar={this.fechaErro}
            objErro={objErro}
          />
          {/* **************************************************************************** */}
          
          <View style={styles.viewTitulo}>
            <Text style={styles.titulo}>Aposentadoria</Text>
          </View>

          <View style={styles.viewVertical}>
            <View style={styles.viewHorizontal}>
              <View style={styles.viewLogo}>
                <Image 
                  style={styles.imgLogo}
                  source={star}
                />
              </View>
              <View style={styles.viewPosLogo}>
                <Text style={styles.aposent_vinheta}>Nossa sugestão para começar agora!</Text>
              </View>
            </View>
          </View>
          <View style={styles.viewVertical}>
            <View style={styles.viewHorizontal}>
              <View style={styles.aposent_viewCalculo}>
                <Text style={[styles.label, styles.aposent_lbCalculo]}>Sua faixa estária:</Text>
                <Text style={[styles.label, styles.aposent_lbCalculo]}>{this.faixaEtaria()}</Text>
              </View>
              <View style={styles.aposent_viewCentral}>
                <Text style={styles.aposent_txDireita}>
                  {`${monetizar(this.rendaPercentual())} (${this.sugestaoReserva() * 100}%)`}
                </Text>
              </View>
            </View>
          </View>

          <View style={styles.separador} />
          
          <View style={styles.viewCentral}>
            <Text style={styles.aposent_vinheta}>Faça seu cálculo</Text>
          </View>

          <LimiteDeErro>
            <SliderDisponib 
              inicial={this.state.disponib}
              retorno={this.defDisponib} 
            />
          </LimiteDeErro>

          <View style={styles.viewVertical}>
            <Text style={styles.label}>Reserva existente em plano previdenciário privado:</Text>
              <TextInput
                ref='reservaPrev'
                style={styles.aposent_input}
                keyboardType='numeric'
                maxLength={10}
                autoCorrect={false}
                selectTextOnFocus
                underlineColorAndroid='#EAEAEA'
                onChangeText={(text) => this.setState({ reservaPrev: desmonetizar(text) })}
                value={monetizar(this.state.reservaPrev)}
              />
          </View>

          <View style={styles.viewCentral}>
            <Text style={styles.aposent_vinheta}>Simulação</Text>
          </View>

          <SliderIdade 
            inicial={this.state.idadeAposent}
            retorno={this.defIdadeAposent} 
          />

          <SliderTaxa 
            inicial={this.state.rentab}
            retorno={this.defRentab} 
          />
          
          <View style={styles.separador} />

          <View style={styles.viewVertical}>
            <View style={styles.viewHorizontal}>
              <View style={styles.aposent_viewReserva}>
                <Text style={[styles.label, styles.aposent_lbReserva]}>Reserva total:</Text>
              </View>
              <View style={styles.aposent_viewReserva}>
                <Text style={styles.aposent_txResTotal}>{monetizar(this.reservaTotal())}</Text>
              </View>
            </View>
          </View>

          <View style={styles.separador} />

        </ScrollView>

        <Rodape
          valor={this.state.comprometimento}
          funcProxTela={this.proxTela}
          tela='Seguranca'
        />
        
      </View>
    );
  }
}
