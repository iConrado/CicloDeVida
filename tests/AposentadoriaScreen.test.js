import 'react-native';
import renderer from 'react-test-renderer';

import React from 'react';
import AposentadoriaScreen from '../src/components/AposentadoriaScreen';


it('renders correctly', () => {
  const rendered = renderer.create(<AposentadoriaScreen />).toJSON();
  expect(rendered).toMatchSnapshot();
});
