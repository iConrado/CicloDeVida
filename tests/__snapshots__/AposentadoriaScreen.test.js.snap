// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "flex-start",
        "justifyContent": "space-between",
        "paddingBottom": 20,
        "paddingHorizontal": 10,
        "paddingTop": 5,
      }
    }
    style={
      Object {
        "flexShrink": 1,
      }
    }
  >
    <View>
      <Modal
        animationType="fade"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        transparent={true}
        visible={false}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#000",
                "flex": 1,
                "height": "100%",
                "justifyContent": "center",
                "opacity": 0.8,
                "position": "absolute",
                "width": "100%",
                "zIndex": 1,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#FFF",
                "borderRadius": 10,
                "justifyContent": "center",
                "paddingHorizontal": 5,
                "paddingVertical": 10,
                "position": "absolute",
                "width": "80%",
                "zIndex": 2,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 14,
                  "marginBottom": 5,
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 14,
                }
              }
            />
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "backgroundColor": "#5C86F5",
                  "borderRadius": 2,
                  "elevation": 4,
                  "marginTop": 10,
                  "minWidth": "30%",
                  "opacity": 1,
                  "paddingVertical": 4,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#FFF",
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginHorizontal": 5,
                  }
                }
              >
                OK
              </Text>
            </View>
          </View>
        </View>
      </Modal>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginBottom": 5,
            "width": "100%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#14567A",
              "fontSize": 20,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Aposentadoria
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginBottom": 8,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "backgroundColor": "#E7A730",
                "borderRadius": 12,
                "height": 24,
                "justifyContent": "center",
                "width": 24,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/imgs/ic_stars_white.png",
                }
              }
              style={
                Object {
                  "height": 24,
                  "width": 24,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "paddingLeft": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#666",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Nossa sugestão para começar agora!
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginBottom": 8,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
                "justifyContent": "center",
                "width": "70%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#666",
                  },
                  Object {
                    "alignSelf": "center",
                    "textAlign": "center",
                    "textAlignVertical": "center",
                  },
                ]
              }
            >
              Sua faixa estária:
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#666",
                  },
                  Object {
                    "alignSelf": "center",
                    "textAlign": "center",
                    "textAlignVertical": "center",
                  },
                ]
              }
            >
              Idade inválida ou não definida
            </Text>
          </View>
          <View
            style={
              Object {
                "height": 40,
                "justifyContent": "center",
                "width": "30%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "textAlign": "center",
                  "textAlignVertical": "center",
                  "width": "100%",
                }
              }
            >
              R$ 0 (0%)
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#14567A",
            "borderBottomWidth": 1,
            "width": "100%",
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#666",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Faça seu cálculo
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginBottom": 8,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#666",
              }
            }
          >
            Sua disponibilidade mensal: 
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <View
            animationType="timing"
            onLayout={[Function]}
            onSlidingComplete={[Function]}
            onValueChange={[Function]}
            step={50}
            style={
              Array [
                Object {
                  "height": 40,
                  "justifyContent": "center",
                },
                Object {
                  "height": 40,
                  "width": "70%",
                },
              ]
            }
            thumbTouchSize={
              Object {
                "height": 40,
                "width": 40,
              }
            }
            value={0}
          >
            <View
              onLayout={[Function]}
              renderToHardwareTextureAndroid={true}
              style={
                Array [
                  Object {
                    "backgroundColor": "#b3b3b3",
                  },
                  Object {
                    "borderRadius": 2,
                    "height": 4,
                  },
                  undefined,
                ]
              }
            />
            <View
              collapsable={undefined}
              renderToHardwareTextureAndroid={true}
              style={
                Object {
                  "backgroundColor": "#14567A",
                  "borderRadius": 2,
                  "height": 4,
                  "opacity": 0,
                  "position": "absolute",
                  "width": 0,
                }
              }
            />
            <View
              collapsable={undefined}
              onLayout={[Function]}
              renderToHardwareTextureAndroid={true}
              style={
                Object {
                  "backgroundColor": "#14567A",
                  "borderRadius": 10,
                  "height": 20,
                  "opacity": 0,
                  "position": "absolute",
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                  ],
                  "width": 20,
                }
              }
            />
            <View
              onMoveShouldSetResponder={[Function]}
              onMoveShouldSetResponderCapture={[Function]}
              onResponderEnd={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderReject={[Function]}
              onResponderRelease={[Function]}
              onResponderStart={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onStartShouldSetResponderCapture={[Function]}
              renderToHardwareTextureAndroid={true}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {},
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "height": 40,
                "justifyContent": "center",
                "width": "30%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "textAlign": "center",
                  "textAlignVertical": "center",
                  "width": "100%",
                }
              }
            >
              0
              %
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "textAlign": "center",
                  "textAlignVertical": "center",
                  "width": "100%",
                }
              }
            >
              R$ 0
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginBottom": 8,
            "width": "100%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#666",
            }
          }
        >
          Reserva existente em plano previdenciário privado:
        </Text>
        <TextInput
          allowFontScaling={true}
          autoCorrect={false}
          keyboardType="numeric"
          maxLength={10}
          onChangeText={[Function]}
          selectTextOnFocus={true}
          style={
            Object {
              "borderBottomColor": "#AAA",
              "borderBottomWidth": 1,
              "height": 30,
              "paddingHorizontal": 5,
              "width": "100%",
            }
          }
          underlineColorAndroid="#EAEAEA"
          value="R$ 0"
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#666",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Simulação
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginBottom": 8,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#666",
              }
            }
          >
            Idade desejada para aposentadoria (em anos):
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <View
            animationType="timing"
            onLayout={[Function]}
            onSlidingComplete={[Function]}
            onValueChange={[Function]}
            step={1}
            style={
              Array [
                Object {
                  "height": 40,
                  "justifyContent": "center",
                },
                Object {
                  "height": 40,
                  "width": "70%",
                },
              ]
            }
            thumbTouchSize={
              Object {
                "height": 40,
                "width": 40,
              }
            }
            value={50}
          >
            <View
              onLayout={[Function]}
              renderToHardwareTextureAndroid={true}
              style={
                Array [
                  Object {
                    "backgroundColor": "#b3b3b3",
                  },
                  Object {
                    "borderRadius": 2,
                    "height": 4,
                  },
                  undefined,
                ]
              }
            />
            <View
              collapsable={undefined}
              renderToHardwareTextureAndroid={true}
              style={
                Object {
                  "backgroundColor": "#14567A",
                  "borderRadius": 2,
                  "height": 4,
                  "opacity": 0,
                  "position": "absolute",
                  "width": 0,
                }
              }
            />
            <View
              collapsable={undefined}
              onLayout={[Function]}
              renderToHardwareTextureAndroid={true}
              style={
                Object {
                  "backgroundColor": "#14567A",
                  "borderRadius": 10,
                  "height": 20,
                  "opacity": 0,
                  "position": "absolute",
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                  ],
                  "width": 20,
                }
              }
            />
            <View
              onMoveShouldSetResponder={[Function]}
              onMoveShouldSetResponderCapture={[Function]}
              onResponderEnd={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderReject={[Function]}
              onResponderRelease={[Function]}
              onResponderStart={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onStartShouldSetResponderCapture={[Function]}
              renderToHardwareTextureAndroid={true}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {},
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "height": 40,
                "justifyContent": "center",
                "width": "30%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "textAlign": "center",
                  "textAlignVertical": "center",
                  "width": "100%",
                }
              }
            >
              50
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginBottom": 8,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#666",
              }
            }
          >
            Taxa de rentabilidade estimada (a.a.):
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <View
            animationType="timing"
            onLayout={[Function]}
            onSlidingComplete={[Function]}
            onValueChange={[Function]}
            step={0.1}
            style={
              Array [
                Object {
                  "height": 40,
                  "justifyContent": "center",
                },
                Object {
                  "height": 40,
                  "width": "70%",
                },
              ]
            }
            thumbTouchSize={
              Object {
                "height": 40,
                "width": 40,
              }
            }
            value={0}
          >
            <View
              onLayout={[Function]}
              renderToHardwareTextureAndroid={true}
              style={
                Array [
                  Object {
                    "backgroundColor": "#b3b3b3",
                  },
                  Object {
                    "borderRadius": 2,
                    "height": 4,
                  },
                  undefined,
                ]
              }
            />
            <View
              collapsable={undefined}
              renderToHardwareTextureAndroid={true}
              style={
                Object {
                  "backgroundColor": "#14567A",
                  "borderRadius": 2,
                  "height": 4,
                  "opacity": 0,
                  "position": "absolute",
                  "width": 0,
                }
              }
            />
            <View
              collapsable={undefined}
              onLayout={[Function]}
              renderToHardwareTextureAndroid={true}
              style={
                Object {
                  "backgroundColor": "#14567A",
                  "borderRadius": 10,
                  "height": 20,
                  "opacity": 0,
                  "position": "absolute",
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateY": 0,
                    },
                  ],
                  "width": 20,
                }
              }
            />
            <View
              onMoveShouldSetResponder={[Function]}
              onMoveShouldSetResponderCapture={[Function]}
              onResponderEnd={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderReject={[Function]}
              onResponderRelease={[Function]}
              onResponderStart={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onStartShouldSetResponderCapture={[Function]}
              renderToHardwareTextureAndroid={true}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {},
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "height": 40,
                "justifyContent": "center",
                "width": "30%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "textAlign": "center",
                  "textAlignVertical": "center",
                  "width": "100%",
                }
              }
            >
              0,0
              %
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#14567A",
            "borderBottomWidth": 1,
            "width": "100%",
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginBottom": 8,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
                "justifyContent": "center",
                "paddingTop": 5,
                "width": "50%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#666",
                  },
                  Object {
                    "alignSelf": "center",
                    "fontWeight": "bold",
                    "height": 30,
                    "textAlign": "center",
                    "textAlignVertical": "center",
                  },
                ]
              }
            >
              Reserva total:
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
                "justifyContent": "center",
                "paddingTop": 5,
                "width": "50%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "alignSelf": "center",
                  "fontWeight": "bold",
                  "height": 30,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                }
              }
            >
              R$ 0
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#14567A",
            "borderBottomWidth": 1,
            "width": "100%",
          }
        }
      />
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderTopColor": "#D0D0D0",
        "borderTopWidth": 1,
        "flexDirection": "row",
        "height": 60,
        "marginTop": 0,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 6,
          "flexDirection": "row",
          "paddingHorizontal": 8,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 7,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#666",
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Comprometimento de renda atual:
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 3,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#666",
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          0
          %
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 4,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#5C86F5",
            "borderRadius": 2,
            "elevation": 4,
            "height": 35,
            "justifyContent": "center",
            "maxWidth": "90%",
            "minWidth": "80%",
            "opacity": 1,
            "paddingVertical": 4,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#FFF",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginHorizontal": 5,
            }
          }
        >
          PRÓXIMA ETAPA
        </Text>
      </View>
    </View>
  </View>
</View>
`;
